WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/DashboardCardsViewModelSlice.kt

```diff
@@ -0,0 +1,223 @@
+        plansCard: MySiteCardAndItem.Card.DashboardPlansCard?,
+        personalizeCard: MySiteCardAndItem.Card.PersonalizeCardModel?,
+        jpFullInstallFullPlugin: MySiteCardAndItem.Card.JetpackInstallFullPluginCard?,
+        domainRegistrationCard: MySiteCardAndItem.Card.DomainRegistrationCard?,
+    ): List<MySiteCardAndItem> {
+        val cards = mutableListOf<MySiteCardAndItem>()
+        quicklinks?.let { cards.add(it) }
+        quickStart?.let { cards.add(it) }
+        domainRegistrationCard?.let { cards.add(it) }
+        bloganuaryNudgeCard?.let { cards.add(it) }
+        bloggingPromptCard?.let { cards.add(it) }
+        blazeCard?.let { cards.add(it) }
+        cardsState?.let {
+            when (cardsState) {
+                is CardsState.Success -> cards.addAll(cardsState.cards)
+                is CardsState.ErrorState -> cards.add(cardsState.error)
+            }
+        }
+        migrationSuccessCard?.let { cards.add(it) }
+        plansCard?.let { cards.add(it) }
+        jpFullInstallFullPlugin?.let { cards.add(it) }
+        noCardsMessageViewModelSlice.buildNoCardsMessage(cards)?.let { cards.add(it) }
+        personalizeCard?.let { cards.add(it) }
+        return cards.toList()
+    }
+
+    fun initialize(scope: CoroutineScope) {
+        this.scope = scope
+        blazeCardViewModelSlice.initialize(scope)
+        bloggingPromptCardViewModelSlice.initialize(scope)
+        bloganuaryNudgeCardViewModelSlice.initialize(scope)
+        personalizeCardViewModelSlice.initialize(scope)
+        quickLinksItemViewModelSlice.initialization(scope)
+        cardViewModelSlice.initialize(scope)
+        quickStartCardViewModelSlice.initialize(scope)
+    }
+
+    private fun buildCards(site: SiteModel) {
+        jpMigrationSuccessCardViewModelSlice.buildCard()
+        jetpackInstallFullPluginCardViewModelSlice.buildCard(site)
+        blazeCardViewModelSlice.buildCard(site)
+        bloggingPromptCardViewModelSlice.buildCard(site)
+        bloganuaryNudgeCardViewModelSlice.buildCard()
+        personalizeCardViewModelSlice.buildCard()
+        quickLinksItemViewModelSlice.buildCard(site)
+        plansCardViewModelSlice.buildCard(site)
+        cardViewModelSlice.buildCard(site)
+        quickStartCardViewModelSlice.build(site)
+    }
+
+    fun onResume() {
+        selectedSiteRepository.getSelectedSite()?.let {
+            if(showDashboardCards(it))buildCards(it)
+            else uiModel.postValue(emptyList())
+        }
+    }
+
+    fun onSiteChanged() {
+        selectedSiteRepository.getSelectedSite()?.let {
+            if(showDashboardCards(it))buildCards(it)
+            else uiModel.postValue(emptyList())
+        }
+    }
+
+    fun onRefresh() {
+        selectedSiteRepository.getSelectedSite()?.let {
+            if(showDashboardCards(it))buildCards(it)
+            else uiModel.postValue(emptyList())
+        }
+    }
+
+    fun onCleared() {
+        uiModel
```

Comment:
@AjeshRPai - What is this doing here? Do you want to clear the uiModel?

Reply:
Sorry about that, I don't think we need to clear it. So removed that line in ae70da2cb5862bb8179d500b249682b0b8595780